### ISSUE TITLE
Client being detected as a browser by macOS

### ISSUE THREAD

**Issue by moeenio** [gamejolt/next-issue-tracker#1386]
In the macOS settings, the Game Jolt Client shows up as a browser in the default browser settings.
![Screen Shot 2020-05-04 at 11 39 08](https://user-images.githubusercontent.com/37651007/80953483-ea928500-8dfb-11ea-86ff-e847fdda0e51.png)

**Comment by Fancy2209** [comment]
Technically it is a browser, it even Plays youtube

**Comment by moeenio** [comment]
yes, but it's meant only for Gamejolt, it should not be able to be used for opening other webpages

**Comment by Fancy2209** [comment]
Yeah, thast's true.

**Comment by moeenio** [comment]
And yes, it is a browser, if I'm not wrong it's based off Electron (which is basically Chromium)

**Comment by Fancy2209** [comment]
It's based of nw.js, I discover that with a issue I had

**Comment by moeenio** [comment]
Yes

**Comment by Tricky1975** [comment]
The Game Jolt Browser is only a a JavaScript wrapper. It's just a browser which can run any site, it's only configured not to do so, This is not really a bug or anything. It's just that MacOS X is not smart enough to realize what the client truly is. I'm not sure if there's a way to get around this. Apple is pretty paranoid these days, after all.

**Comment by moeenio** [comment]
macOS itself doesn't detect browsers, browsers themselves have to tell macOS they are browsers. This is definitely a bug in the Client.

**Comment by Tricky1975** [comment]
Nope, it isn't. Very likely it isn't. The client itself cannot communicate to the OS like that. This is definitely a MacOS issue detecting the engine the client uses as a browser, and that engine is a browser, and one that was not written by the Game Jolt crew, but an external party. So whatever story you make up, this does NOT belong on this issue tracker as this is NOT a bug in the client itself. The engine just uses technology from existing browsers also. Possibly MacOS reacted to that. I've seen more apps using this engine being detected as browsers by MacOS, and the way that engine is set up, this is only logical, as it was set up to be used as a "normal browser" in case certain programs written for that engine ask for that. 

I've seen MacOS react rather strangely to stuff like this when I was a Mac user myself for many years. I must say Windows is not any different. I've seen a few games of mine be suggested as apps to open a certain file with... This because, some of the assets of that game were in the same format. The client uses an engine that can open normal sites, and sometimes MacOS responds to that. 

The code inside the client is just the same code as used on the normal site after all...

**Comment by moeenio** [comment]
I agree that the bug is from the engine (nw.js). However, the app has to declare itself as a browser/for opening HTML documents (case here) so macOS proposes it as a browser.

**Comment by Tricky1975** [comment]
It's actually not a bug. I'm quite sure that nw.js was deliberately coded to act that way, due to its prime purpose. 
It's one of the (countless) reasons that I cast nw.js away as a viable options for my future projects. Since the client is nothing more but a browser which can communicate with a secret program Fernando coded in Go which a website cannot do from a normal browser (due to security reasons). It's this secret program actually that is the true client, that program manages all installations and runs and all that stuff. The nw.js application is nothing more but the site itself, identical to the site you'd get if you use a normal browser, however the script are able to tell it runs within nw.js and access the features that cannot be accessed when loaded in a "true" browser. I guess MacOS X was merely fooled by this entire setup. 
If the client was a program on its own entirely written in Go or C++ or whatever, then we'd have a different story, but I hope you agree that this was not doable... not even by a big company, let alone by a small group that the crew of GJ actually is.
